docs/meshlets.md

```diff
@@ -0,0 +1,159 @@
+Ownership of the model **must not** be taken within the body of the custom `aMeshletDivision` function. 
+
+Please note that the model will be assigned to each [`meshlet`](../framework/include/meshlet_helpers.hpp#L7) after `aMeshletDivision` has executed (see implementation of [`gvk::divide_indexed_geometry_into_meshlets()`](../framework/include/meshlet_helpers.hpp#L139)).
+
+The custom division function must follow a specific declaration schema. Optional parameters can be omitted, but all of them need to be provided in the following order:
+
+| Parameter | Mandatory? | Description |
+| :------ | :---: | :--- |
+| `const std::vector<glm::vec3>& tVertices` | no | The vertices of the mesh, or of combined meshes of the model |
+| `const std::vector<uint32_t>& tIndices` | yes | The indices of the mesh, or of combined meshes of the model | 
+| `const model_t& tModel` | yes	| The model these meshlets are generated from | 
+| `std::optional<mesh_index_t> tMeshIndex` | yes | The optional mesh index. If no value is passed, it means the meshes of the model are combined into a single vertex and index buffer. | 
+| `uint32_t tMaxVertices` | yes | The maximum number of vertices that should be added to a single meshlet | 
+| `uint32_t tMaxIndices` | yes | The maximum number of indices that should be added to a single meshlet | 
+
+The custom division function **must** return a `std::vector<meshlet>` by value, containing the generated meshlets.
+
+#### Example for Vertices and Indices:
+
+Example of a custom division function that can be passed to `gvk::divide_into_meshlets()`, that takes both, vertices and indices:
+
+```C++
+[](const std::vector<glm::vec3>& tVertices, const std::vector<uint32_t>& aIndices,
+    const gvk::model_t& aModel, std::optional<gvk::mesh_index_t> aMeshIndex,
+    uint32_t aMaxVertices, uint32_t aMaxIndices) {
+
+    std::vector<gvk::meshlet> generatedMeshlets;
+    // Perform meshlet division here, and store resulting meshlet in generatedMeshlets
+    return generatedMeshlets;
+}
+```
+
+#### Example for Indices Only:
+
+Example of a custom division function that can be passed to `gvk::divide_into_meshlets()`, that takes indices only:
+
+```C++
+[](const std::vector<uint32_t>& aIndices,
+    const gvk::model_t& aModel, std::optional<gvk::mesh_index_t> aMeshIndex,
+    uint32_t aMaxVertices, uint32_t aMaxIndices) {
+
+    std::vector<gvk::meshlet> generatedMeshlets;
+    // Perform meshlet division here, and store resulting meshlet in generatedMeshlets
+    return generatedMeshlets;
+}
+```
+
+#### Example That Uses a 3rd Party Library:
+
+An example about how [meshoptimizer](https://github.com/zeux/meshoptimizer) **can** be used via the custom division function that can be passed to `gvk::divide_into_meshlets()`:
+
+Please note: meshoptimizer expects `aMaxIndices` to be divisible by 4. The value 124 has been recommended for Nvidia!
```

Comment:
Please provide references for these two statements (1. divisible by four, 2. 124 recommended for Nvidia)!

Are you sure it is 124 and not 126 (see [Christoph Kubisch - Introduction to Turing Mesh Shaders](https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/)).

If 124 is mentioned somewhere in meshoptimizer, please cite that one!

Reply:
Thanks for the edits.
Sorry bad wording on my part, 124 is the recommendation by meshoptimizer. I'll add that.

Reply:
The divisable by four is not documented in meshoptimizer, I just ran into it while testing.
`assert(max_triangles % 4 == 0); // ensures the caller will compute output space properly as index data is 4b aligned`
This is in the `meshopt_buildMeshlets` function of meshoptimizer.

Reply:
No need to be sorry, this is just polishing. Thanks for adding the citation.